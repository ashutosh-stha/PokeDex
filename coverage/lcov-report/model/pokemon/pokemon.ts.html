
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for model/pokemon/pokemon.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">model/pokemon</a> pokemon.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">70% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>7/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">66.66% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/3</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">70% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>7/10</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {createModel} from '@rematch/core';
import {RootModel} from '../models';
&nbsp;
export interface Pokemon {
  name: string;
  url: string;
}
&nbsp;
interface PokemonResponse {
  results: Pokemon[];
}
&nbsp;
interface PokemonState {
  pokemonList: Pokemon[];
}
&nbsp;
const initialState: PokemonState = {
  pokemonList: [],
};
&nbsp;
export const pokemon = createModel&lt;RootModel&gt;()({
  state: initialState,
  reducers: {
<span class="fstat-no" title="function not covered" >    se</span>tPokemonList(state, payload) {
<span class="cstat-no" title="statement not covered" >      return {...state, pokemonList: payload};</span>
    },
  },
  effects: dispatch =&gt; ({
    async loadPokemon(_, rootState) {
      const apiService = rootState.httpClient.apiService;
      try {
        const pokemonData = await apiService?.get&lt;PokemonResponse&gt;(
          '/pokemon?offset=30&amp;limit=30',
        );
        <span class="missing-if-branch" title="if path not taken" >I</span>if (pokemonData) {
<span class="cstat-no" title="statement not covered" >          dispatch.pokemon.setPokemonList(pokemonData.results);</span>
        }
      } catch (error) {
        // Handle any errors here
<span class="cstat-no" title="statement not covered" >        console.error('Failed to load Pokemon:', error);</span>
      }
    },
  }),
});
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-06-18T10:44:52.181Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    