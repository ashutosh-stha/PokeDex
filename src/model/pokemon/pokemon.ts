import {createModel} from '@rematch/core';
import {RootModel} from '../models';

export interface Pokemon {
  name: string;
  url: string;
}

interface PokemonResponse {
  results: Pokemon[];
}

export type PokemonType = {
  name: string;
};
type Sprites = {
  back_default: string | null;
  back_female: string | null;
  back_shiny: string | null;
  back_shiny_female: string | null;
  front_default: string | null;
  front_female: string | null;
  front_shiny: string | null;
  front_shiny_female: string | null;
  other: {
    dream_world: {
      front_default: string | null;
      front_female: string | null;
    } | null;
    home: {
      front_default: string | null;
      front_female: string | null;
      front_shiny: string | null;
      front_shiny_female: string | null;
    } | null;
    'official-artwork': {
      front_default: string | null;
      front_shiny: string | null;
    } | null;
  };
};

interface PokemonDetail {
  id: number;
  name: string;
  types: {type: PokemonType}[];
  sprites: Sprites;
}

interface PokemonState {
  pokemonList: Pokemon[];
  pokemonDetail?: PokemonDetail;
}

const initialState: PokemonState = {
  pokemonList: [],
  pokemonDetail: undefined,
};
export const pokemon = createModel<RootModel>()({
  state: initialState,
  reducers: {
    setPokemonList(state, payload) {
      return {...state, pokemonList: payload};
    },
    setPokemonDetail(state, payload) {
      return {...state, pokemonDetail: payload};
    },
  },
  effects: dispatch => ({
    async loadPokemon(_, rootState) {
      const apiService = rootState.httpClient.apiService;
      try {
        const pokemonData = await apiService?.get<PokemonResponse>(
          '/pokemon?offset=100&limit=100',
        );
        if (pokemonData) {
          dispatch.pokemon.setPokemonList(pokemonData.results);
        }
      } catch (error) {
        // Handle any errors here
        console.error('Failed to load Pokemon:', error);
      }
    },
    async loadPokemonDetail(pokemonName, rootState) {
      const apiService = rootState.httpClient.apiService;
      try {
        const pokemonDetail = await apiService?.get<PokemonDetail>(
          `/pokemon/${pokemonName}`,
        );
        if (pokemonDetail) {
          dispatch.pokemon.setPokemonDetail(pokemonDetail);
        }
      } catch (error) {
        console.error('Failed to load Pokemon Detail:', error);
      }
    },
  }),
});
