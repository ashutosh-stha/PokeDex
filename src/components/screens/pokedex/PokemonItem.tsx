import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import {Pokemon} from '../../../model/pokemon/pokemon';

interface PokemonItemProps {
  pokemon: Pokemon;
  onPress: () => void;

  selected: Boolean;
}

export const PokemonItem: React.FC<PokemonItemProps> = ({
  pokemon,
  onPress,
  selected,
}) => {
  return (
    <TouchableOpacity
      key={pokemon.name}
      style={[styles.container, selected && styles.selectedContainer]}
      onPress={onPress}>
      <Text style={styles.title}>{pokemon.name}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 8,
    backgroundColor: '#77BA99',
    padding: 10,
    flex: 0.5,
    borderRadius: 30,
  },
  title: {
    textTransform: 'capitalize',
    color: 'white',
    fontSize: 16,
    textAlign: 'center',
  },
  selectedContainer: {
    backgroundColor: '#262730',
  },
});
