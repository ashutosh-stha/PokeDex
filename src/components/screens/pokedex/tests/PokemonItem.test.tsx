import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {PokemonItem, PokemonItemProps} from '../PokemonItem';
import {Pokemon} from '../../../../model/pokemon/pokemon';

const MOCK_POKEMON_DATA: Pokemon = {
  name: 'Pikachu',
  url: 'url',
};

const defaultProps: PokemonItemProps = {
  onPress: jest.fn(),
  selected: false,
  pokemon: MOCK_POKEMON_DATA,
};

const renderWrapper = (props: PokemonItemProps | {}) =>
  render(<PokemonItem {...defaultProps} {...props} />);

describe('PokemonItem', () => {
  it('should render correctly', () => {
    expect(renderWrapper({})).toMatchSnapshot();
  });
  it('when user press the button', () => {
    const {getByTestId} = renderWrapper({});
    fireEvent.press(getByTestId('pokemonBtn'));
    expect(defaultProps.onPress).toHaveBeenCalled();
  });
  it('when the pokemon is selected', () => {
    expect(renderWrapper({selected: true})).toMatchSnapshot();
  });
});
