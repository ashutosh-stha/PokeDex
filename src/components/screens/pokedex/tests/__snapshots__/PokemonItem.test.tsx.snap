// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonItem should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#77BA99",
      "borderRadius": 30,
      "flex": 0.5,
      "margin": 8,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="pokemonBtn"
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 16,
        "textAlign": "center",
        "textTransform": "capitalize",
      }
    }
  >
    Pikachu
  </Text>
</View>
`;

exports[`PokemonItem when the pokemon is selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#262730",
      "borderRadius": 30,
      "flex": 0.5,
      "margin": 8,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="pokemonBtn"
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 16,
        "textAlign": "center",
        "textTransform": "capitalize",
      }
    }
  >
    Pikachu
  </Text>
</View>
`;
